docs/guide/arrays.md

```diff
@@ -0,0 +1,127 @@
+* A result of an arithmetic operation (e.g. `5*A1:B5`)
+* A result of a function (e.g. `=ARRAYFORMULA(ARRAY_CONSTRAIN(A2:E5,2,2))`)
+
+An array is inherently a two-dimensional object.
+
+`1`x`1` arrays are treated as single, zero-dimensional values (**scalars**).
+
+## Enabling the array arithmetic mode
+
+To use arrays in HyperFormula, you need to enable the **array arithmetic mode**.
+
+You can enable the array arithmetic mode:
+* [Locally](#enabling-the-array-arithmetic-mode-locally) (for an individual function or operation)
+* [Globally](#enabling-the-array-arithmetic-mode-globally) (for your HyperFormula instance)
+
+### Enabling the array arithmetic mode locally
+
+To enable the array arithmetic mode once, within a particular function or formula, use the `ARRAYFORMULA` function:
+
+| Syntax | Example |
+| :--- | :--- |
+| `ARRAYFORMULA(your_array_formula)` | `=ARRAYFORMULA(A2:A5*B2:B5)` |
+| `ARRAYFORMULA(YOUR_FUNCTION(your_array_formula))` | `=ARRAYFORMULA(ISEVEN(A2:A5*10))` |
+
+### Enabling the array arithmetic mode globally
+
+To enable the array arithmetic mode by default, everywhere in your HyperFormula instance:
+
+* In your HyperFormula [configuration](../api/interfaces/configparams.html#usearrayarithmetic), set the `useArrayArithmetic` option to `true`.
+
+With the array arithmetic mode enabled globally, you can operate on arrays without using the `ARRAYFORMULA` function:
+
+```
+=A2:A5*B2:B5
+
+ISEVEN(A2:A5*10)
+```
+
+## Array features
+
+HyperFormula lets you:
+* [Operate on arrays](#operating-on-arrays) just like on [scalars](#about-arrays)
+* [Pass arrays to functions](#passing-arrays-to-scalar-functions) that accept [scalars](#about-arrays)
+* [Broadcast](#broadcasting) a smaller input array across a larger output area
+* [Filter an array](#filtering-an-array) based on boolean arrays
+* [Constrain](#constraining-an-array-s-size) an array's size
+
+### Operating on arrays
+
+You can operate on arrays just like on single values.
+
+With the [array arithmetic mode](#enabling-the-array-arithmetic-mode) enabled, each output array value is the result of your operation on the corresponding input array value.
+
+```
+=ARRAYFORMULA(A2:A5*B2:B5)
+```
+
+### Passing arrays to scalar functions
+You can pass arrays to functions that would normally accept [scalars](#about-arrays) (thanks to HyperFormula's **vectorization** feature).
+
+When you pass an array to a function that accepts [scalars](#about-arrays), that function produces an array on the output as well (with the [array arithmetic mode](#enabling-the-array-arithmetic-mode) enabled).
+
+```
+=ARRAYFORMULA(ISEVEN(A2:A5*10))
+```
+
+### Broadcasting
+You can broadcast a smaller input array across a larger output area (thanks to HyperFormula's **broadcasting** feature).
+
+With the [array arithmetic mode](#enabling-the-array-arithmetic-mode) enabled, the smaller array is automatically “broadcast” across the larger array so that both arrays have the same dimensions.
```

Comment:
broadcasting is only specific to vectorization, does not happen outside of vectorization

also, broadcasting happens only if particular dimension is `1`. Otherwise, the smaller dimension is enlarged to the larger one, but missing values are filled with NA's

Reply:
@izulin - done: 5e3e2d2